strr-examiner-web/app/components/Host/EditRentalUnitForm.vue

```diff
@@ -0,0 +1,267 @@
+<script setup lang="ts">
+
+const exStore = useExaminerStore()
+const {
+  rentalUnitAddressToEdit,
+  hasUnsavedRentalUnitChanges,
+  rentalUnitAddressSchema
+} = storeToRefs(exStore)
+const strrModal = useStrrModals()
+const { t } = useI18n()
+const confirmUnchangedModal = ref<ConfirmModal | null>(null)
+
+const route = useRoute()
+const isApplicationRoute = computed(() => route.path.includes('/examine/'))
+const isRegistrationRoute = computed(() => route.path.includes('/registration/'))
+const { isFeatureEnabled } = useFeatureFlags()
+const canEditApplicationAddress = isFeatureEnabled('enable-examiner-edit-address-application')
+
+const routeId = computed(() => {
+  if (isApplicationRoute.value) {
+    return route.params.applicationId as string
+  } else if (isRegistrationRoute.value) {
+    return route.params.registrationId as string
+  }
+  return null
+})
+
+const currentAddress = ref({ ...rentalUnitAddressToEdit.value })
+const editStrAddressform = ref()
+const isLoading = ref(false)
+
+watch(currentAddress, () => {
+  const originalKeys = Object.keys(rentalUnitAddressToEdit.value)
+  const currentKeys = Object.keys(currentAddress.value)
+  const allKeys = [...new Set([...originalKeys, ...currentKeys])]
+
+  const hasChanges = allKeys.some((key) => {
```

Comment:
You can use lodash `isEqual()` to compare.

Reply:
Done